FrontEnd/src/app/components/users/chat/room/room.component.ts

```diff
@@ -0,0 +1,259 @@
+import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router, ParamMap } from '@angular/router';
+import { Rooms } from 'src/app/Models/ChatModels/Rooms';
+import { RoomUsers } from 'src/app/Models/ChatModels/RoomUsers';
+import { UserReceived } from 'src/app/Models/UsersReceived';
+import { AuthService } from 'src/app/services/auth.service';
+import { ChatService } from 'src/app/services/chat.service';
+import { LocalStorageService } from 'ngx-webstorage';
+import { colors } from 'src/app/components/users/chat/colors';
+import { map, take } from 'rxjs/operators';
+import { MessageReceived } from 'src/app/Models/Messages';
+import { ConnectedUsers } from 'src/app/Models/ChatModels/ConnectedUsers';
+
+@Component({
+  selector: 'app-room',
+  templateUrl: './room.component.html',
+  styleUrls: ['./room.component.scss'],
+})
+export class RoomComponent implements OnInit {
+  object = this.auth.decryptedAndDecodedToken();
+  loading: boolean;
+  singleRoom: Rooms;
+  roomMessages: MessageReceived[];
+  user: UserReceived;
+  usersList: RoomUsers[];
+  showUsers: RoomUsers[];
+  sendMessageRoom: FormGroup;
+  rooms: Rooms[] = [];
+  showRooms: Rooms[] = [];
+  roomId: string;
+  onlineUsers: ConnectedUsers[] = [];
+
+  @ViewChild('message') messageRef: ElementRef;
+  @ViewChild('container') containerRef: ElementRef;
+  constructor(
+    private chatService: ChatService,
+    private route: ActivatedRoute,
+    private auth: AuthService,
+    private LocalStorageService: LocalStorageService,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.chatService.retrieveRoom().subscribe((room: Rooms) => {
+      this.showRooms.push(room);
+    });
+
+    this.chatService.retrieveNewMessage().subscribe((message: any) => {
+      if (this.roomId == message.id) {
+        this.addMessageToBox(message);
+      }
+      setTimeout(() => {
+        this.containerRef.nativeElement.scrollTop = this.containerRef.nativeElement.scrollHeight;
+      }, 200);
+    });
+
+    this.chatService.removeUser().subscribe((user) => {
+      this.switchOffline(user.id);
+    });
+
+    this.chatService
+      .GetAllRoom()
+      .pipe(take(1))
+      .subscribe((rooms: Rooms[]) => {
+        this.rooms = rooms;
+        this.showRooms = [...this.rooms];
+      });
+
+    this.route.params.subscribe((params) => {
+      this.roomId = params.id;
+      this.chatService
```

Comment:
see combineLatest RxJs function to avoid nested subscriptions : https://www.learnrxjs.io/learn-rxjs/operators/combination/combinelatest.